activesupport/lib/active_support/cache/mem_cache_store.rb

```diff
@@ -102,26 +102,28 @@ def initialize(*addresses)
 
       # Increment a cached value. This method uses the memcached incr atomic
       # operator and can only be used on values written with the :raw option.
-      # Calling it on a value not stored with :raw will initialize that value
-      # to zero.
+      # Calling it on a value not stored with :raw will fail and return nil.
+      # With the :initial option, its vlaue is used for a new counter.
+      # Otherwise, an operation to a nonexistant value will fail and return nil.
       def increment(name, amount = 1, options = nil)
         options = merged_options(options)
         instrument(:increment, name, amount: amount) do
           rescue_error_with nil do
-            @data.with { |c| c.incr(normalize_key(name, options), amount, options[:expires_in]) }
+            @data.with { |c| c.incr(normalize_key(name, options), amount, options[:expires_in], options[:initial]) }
           end
         end
       end
 
       # Decrement a cached value. This method uses the memcached decr atomic
       # operator and can only be used on values written with the :raw option.
-      # Calling it on a value not stored with :raw will initialize that value
-      # to zero.
+      # Calling it on a value not stored with :raw will fail and returns nil.
+      # With the :initial option, its vlaue is used for a new counter.
+      # Otherwise, an operation to a nonexistant value will fail and return nil.
```

Comment:
```suggestion
      # Calling it on a value not stored with :raw will fail and return nil.
      # With the :initial option, its value is used for a new counter.
      # Otherwise, an operation to a nonexistent value will fail and return nil.
```

Reply:
Applied, thanks